### PR TITLE
Clean up the integration tests

### LINKED ISSUE #3698: Segfault in `wasmer run-unstable`'s integration tests on musl linux
### Describe the bug

The integration tests for `wasmer run-unstable` (#3650) are currently segfaulting... but only when run in CI and building for `x86_64-unknown-linux-musl`.

This is probably related to WASIX or our integration testing and not the `wasmer run-unstable` PR.

### Steps to reproduce

Push a commit to GitHub and watch the musl linux integration tests fail ([example](https://github.com/wasmerio/wasmer/actions/runs/4471026419/jobs/7855861497)).

You should see the following:

```
...
error: test failed, to rerun pass '-p wasmer-integration-tests-cli --test run_unstable'

Caused by:
  process didn't exit successfully: `/__w/wasmer/wasmer/target/x86_64-unknown-linux-musl/debug/deps/run_unstable-2b992cbc0342feb2 --nocapture --test-threads=1` (signal: 11, SIGSEGV: invalid memory reference)
```

### Additional context

To make sure #3650 is mergeable, we've disabled the tests in xxxx.

### LINKED ISSUE #4015: The `wasmer-integration-tests-cli` test suite isn't hygienic
### Motivation

Almost all of the tests in the `wasmer-integration-tests-cli` test suite are non-hygienic. They all use the same `~/.wasmer/` directory, which means that

1. Integration tests need to be run sequentially (i.e. `--test-threads=1`) and a failure midway through can sometimes mess things up for subsequent tests (I've noticed caching is particularly bad for this)
2. After the integration tests are done, the global `~/.wasmer/wasmer.toml` file will be left in a broken state and you'll need to find your old access token and log in again

### Proposed solution

The way we solve this problem is by making sure each integration test uses its own independent wasmer directory. We can do this fairly trivially by setting the `$WASMER_DIR` environment variable to something unique like [`$CARGO_TARGET_TMPDIR/wasmer-integration-tests-cli/$test_name`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) 

Cargo always sets provides integration tests with a `$CARGO_TARGET_TMPDIR` directory inside `target/` they can use as a scratch space. That also means CI will get caching for free.

### PR DESCRIPTION
**Pull request by Michael-F-Bryan**
This PR does a couple things to make our CLI integration test story a bit nicer:

- The run-unstable integration tests have been merged into `run.rs` because `wasmer run-unstable` no longer exists
- We now use the `--registry` flag from #4013 instead of running `wasmer login` and clobbering the tokens in the user's `$WASMER_DIR/wasmer.toml` (Fixes #4015)
- The run-unstable tests now run the `wasmer` executable directly instead of making a defensive `cargo run --features=...` 
  - The `cargo run` doesn't make sense any more because #4020 decided that we'll rely on `make integration-test-cli` or the user to provide an up-to-date `wasmer` executable
- Some create-exe tests were moved from `run.rs` into `create_exe.rs` where they belong 
- Moved the paths for all files used during testing into a single `fixtures` module
- Remove the `--test-threads=1` from `make test-integration-cli` so we use all cores when running integration tests
- Remove the `--no-capture` from `make test-integration-cli` to reduce the amount of spam you are bombarded with when running tests

After these changes, the *"Test Integration CLI"* CI step is reduced from taking about 30 minutes to 3 minutes.

It also looks like the MUSL segfaults are gone (i.e. fixes #3698).

### CODE REVIEW

File: tests/integration/cli/tests/run.rs
Comment (ptitSeb) [inline tests/integration/cli/tests/run.rs:289]: This works? Why it doesn't when I try on php/php https://github.com/wasmerio/wasmer/issues/4027, is there something special to do on package itself?

File: tests/integration/cli/tests/run.rs
Comment (ptitSeb) [inline tests/integration/cli/tests/run.rs:356]: still targetting "wapm.io"?

**Review by ptitSeb — APPROVED** [review]

File: tests/integration/cli/src/assets.rs
Comment (ptitSeb) [inline tests/integration/cli/src/assets.rs:24]: When building from the CLI, the path might be `target/x86_64-linux/release` or something similar as we define the target arch even when not cross-compiling

File: tests/integration/cli/tests/run.rs
Comment (Michael-F-Bryan) [reply tests/integration/cli/tests/run.rs:356]: Ah good point, I'll go through and update everything to use either `wasmer.io` or `wasmer.wtf`.

File: tests/integration/cli/tests/run.rs
Comment (Michael-F-Bryan) [reply tests/integration/cli/tests/run.rs:290]: I'm not entirely sure why it doesn't work. There are loads of tests that use `--mapdir` so directory mapping definitely works. You can even mount to `/`, but it'll break packages that bundle their own files (e.g. python and its standard library). 

Maybe it's an issue with PHP's filesystem access?

File: Makefile
Comment (syrusakbary) [inline Makefile:630]: Nice!

**Comment by Michael-F-Bryan** [comment]
I'm going to put this PR on hold until #4050 is merged. We're seeing a bunch of new runtime-in-runtime issues during the snapshot tests, which doesn't really make sense this PR hasn't touched any code in the main `wasmer-wasix` crate.

**Pull request merged**
